streams/src/main/java/org/apache/kafka/streams/errors/StreamsUncaughtExceptionHandler.java

```diff
@@ -27,6 +27,7 @@
      * Enumeration that describes the response from the exception handler.
      */
     enum StreamThreadExceptionResponse {
+        REPLACE_THREAD(0, "REPLACE_STREAM_THREAD"),
```

Comment:
I would rename it to `REPLACE_STREAM_THREAD`.

Reply:
Oh, good catch.

Just a quick question: did we misname this option in the KIP? A StreamThread is a specific kind of thread. What I mean is that a GlobalStreamThread is _not_ a StreamThread. Perhaps `REPLACE_THREAD` and `SHUTDOWN_THREAD` would have been better, more general names, for these. If you agree, I think we can just amend the KIP and fix it in this PR.

Reply:
It should be the string is `REPLACE_THREAD`. Ill just fix that, its just left over from a period of time where we thought of calling it `REPLACE_STREAM_THREAD`

Reply:
I actually would be in favor of calling the enum value `REPLACE_STREAM_THREAD`. A stream thread is a stream thread and a global stream thread is a global stream thread. I am aware that the KIP calls the enum value differently, but we also have a config that is called 'NUM_STREAM_THREADS_CONFIG' and we have also 'addStreamThread()' and `removeStreamThread()`. So I guess, the name to the outside of this is stream thread and not thread. We have also other threads in Kafka Streams like the state directory cleaner thread and the RocksDB metrics recording thread.

Reply:
What if we add an option to replace the global thread?

Reply:
Ah, now I got it! Sorry! Makes sense! In that case we can reuse `REPLACE_THREAD` also for the global stream thread. Forgot about that!